docs: edit lifecycle management getting started (#2602)
keptn/lifecycle-toolkit: docs/content/en/docs/getting-started/lifecycle-management.md

> This tutorial assumes you have already completed the
[Getting started with Keptn Observability](../getting-started/)
exercise.
> Please ensure you've finished that before attempting this exercise.

## Keptn Pre and Post Deployment Tasks

When Keptn is successfully monitoring your deployments,
it can also run arbitrary tasks and SLO evaluations for you
either pre-deployment or post-deployment
as specified in the pod template specs of your
[Workloads](https://kubernetes.io/docs/concepts/workloads/)
([Deployments](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/),
[StatefulSets](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/),
[DaemonSets](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/),
and
[ReplicaSets](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/).

> Pre and post deployments can also run on a `KeptnApp` level.

> > A `KeptnApp` concatenates multiple workloads into a single application.

Reviewer: ```suggestion
> A `KeptnApp` groups multiple workloads into a single application.
```

### Verify it works

> Verify that the `KeptnTaskDefinition` resource shown above above actually works.

Reviewer: ```suggestion
Verify that the `KeptnTaskDefinition` resource shown above actually works.
```

Trigger an on-demand task execution

> to verify that the job and pod function correctly.

Reviewer: ```suggestion
to verify that the job and pod are working correctly.
```

![webhook sync](../assets/webhook.site.3.png)

Incidentally, this is exactly how you can use Keptn with
[applications deployed outside of Kubernetes](../use-cases/non-k8s.md).

> > Note: If you want to trigger this manifest multiple times,

Reviewer: ```suggestion
> Note: If you want to trigger this task multiple times,
```

you must change the value of the `name` field

> in he `KeptnTask` resource each time.

Reviewer 2: ```suggestion
in the `KeptnTask` resource each time.
```

### What Happens Next?

1. The deployment is applied.
1. When the pods are running,
   Keptn automatically creates a `KeptnTask` resource
   for version `0.0.2` of this `KeptnApp`.

> 1. The `KeptnTask` creates a Kubernetes `job`.
> 1. The Kubernetes `job` creates a Kubernetes `pod`.
> 1. The `pod` runs `curl` and sends a new event to the event sink.

Reviewer 2: I think we still capitalize resources that come from k8s right?
```suggestion
1. The `KeptnTask` creates a Kubernetes `Job`.
1. The Kubernetes `Job` creates a Kubernetes `Pod`.
1. The `Pod` runs `curl` and sends a new event to the event sink.
```

### Pre-deployment Tasks

Keptn Tasks can also be executed pre-deployment (before the pods are scheduled).
Do this by using the `keptn.sh/pre-deployment-tasks` label or annotation.

> > Note: If a pre-deployment task fails, the `pod` remains in a Pending state.

Reviewer 2: ```suggestion
> Note: If a pre-deployment task fails, the `Pod` remains in a Pending state.
```
